NodeJS/src/AgentClient.ts

```diff
@@ -0,0 +1,413 @@
+const lodash = require('lodash');
+import {Session} from "./session";
+
+const STATION_NAME = 'generated station';
+const AGENT_FIRST_NAME = 'generated';
+const AGENT_LAST_NAME = 'agent';
+// ten seconds
+const INTERVAL_IN_MILLIS = 10 * 1000;
+const MAX_RETRY = 5;
+
+interface SkillPriority{
+    skillNumber:number,
+    skillPriority:number
+}
+
+class AgentClient {
+    session:Session;
+
+    constructor(session: any) {
+        this.session = session
+    }
+
+    async createAgent(agent_username: string, agent_password: string, skillsWithPriority: [SkillPriority]) {
+        await this.session.login();
+        let subAccountId = await this.getSubAccountId();
+        let agentStationGroupId = await this.getAgentStationGroupId(subAccountId);
+        let agentLoginId = await this.generateExtension(subAccountId, 'AGENT');
+        let skillIds = await this.getSkillIds(subAccountId);
+        await this.sendCreateAgentRequest(
```

Comment:
What does this return?

Reply:
It will return the agent object. But we are not using it here, so we will wait until the actual agent gets created and will retrieve the agent.